instance-scheduler/main_int_test.go

```diff
@@ -42,7 +42,7 @@ func TestHandler(t *testing.T) {
 		assert.Greater(t, res.ActedUpon, 20, fmt.Sprintf("Number of instances acted upon seems too low. %v", addMsg))
 		assert.Less(t, res.ActedUpon, 100, fmt.Sprintf("Number of instances acted upon seems too high. %v", addMsg))
 		assert.Greater(t, res.Skipped, 0, fmt.Sprintf("Number of skipped instances seems too low. %v", addMsg))
-		assert.Less(t, res.Skipped, 50, fmt.Sprintf("Number of skipped instances seems too high. %v", addMsg))
+		assert.Less(t, res.Skipped, 80, fmt.Sprintf("Number of skipped instances seems too high. %v", addMsg))
```

Comment:
Another coupling facet, but paired with arbitrary values - which means it's only a matter of time until this becomes an issue again 💣 Since as the platform grows, so will the number of skipped instances, instances acted upon etc.

I'd favour a general smoke test with full integration rather than using arbitrary values to test, thoughts? 👀

Reply:
I agree on this point, it would be better, if we actually tested on the instances in one of our accounts, but that would mean always having an instance there (unless we test it with terratest) and it would not contribute to test coverage (as we would not want to have all variations of instances/tags/account, so probably run only one instance, although that's OK, as the logic test is covered in unit tests. Another point why this is useful though is that the instance role gets tested here, where if we only test it on one instance, we don't get the confidence that the role config is correct (hence the thresholds in this test).

Reply:
I agree that it does bring value in terms of adding test coverage to aspect of the systems that a notoriously difficult to test 🧪 (such as permissions)

Although it add a lot of complexity and fragility to the CI/CD - arguably diminishing its overall value (but happy for different opinions than my own 😁)

I'd prefer to offload the responsibility of validating these aspects of the system to production runs with appropriate error handling (failing fast) in the code and alerting systems to notify us when something goes wrong 🔥 This would simplify the CI/CD, reduce maintenance and minimise the noise of errors 🙈